src/hpa_central.c

```diff
@@ -189,21 +169,24 @@ hpa_central_dalloc(tsdn_t *tsdn, hpa_central_t *central, edata_t *edata) {
 	edata_addr_set(edata, edata_base_get(edata));
 	edata_zeroed_set(edata, false);
 
-	if (!edata_is_head_get(edata)) {
-		edata_t *lead = hpa_central_dalloc_get_merge_candidate(tsdn,
-		    central, edata_before_get(edata));
-		if (lead != NULL) {
-			eset_remove(&central->eset, lead);
-			hpa_central_dalloc_merge(tsdn, central, lead, edata);
-			edata = lead;
-		}
-	}
-	edata_t *trail = hpa_central_dalloc_get_merge_candidate(tsdn, central,
-	    edata_past_get(edata));
-	if (trail != NULL && !edata_is_head_get(trail)) {
+	/*
+	 *  Merge forward first, so that the original *edata stays active state
+	 *  for the second acquire (only necessary for sanity checking).
+	 */
+	edata_t *trail = emap_try_acquire_edata_neighbor(tsdn, central->emap,
+	    edata, EXTENT_PAI_HPA, extent_state_dirty, true /* forward */);
```

Comment:
I'm not religious about this sort of thing, but everywhere else in the codebase we put the bool name comment before the value rather than after.

Reply:
I'm with you. Will fix.